packages/camera/camera_android_camerax/lib/src/android_camera_camerax.dart

```diff
@@ -419,6 +451,46 @@ class AndroidCameraCameraX extends CameraPlatform {
         targetRotation: targetRotation, targetResolution: targetResolution);
   }
 
+  /// Configures and starts a video recording.
+  @override
+  Future<void> startVideoRecording(int cameraId,
+      {Duration? maxVideoDuration}) async {
+    assert(cameraSelector != null);
+    assert(processCameraProvider != null);
+
+    recorder = Recorder(); // TODO(gmackall): Configure resolution info here.
+    videoCapture = await VideoCapture.withOutput(recorder!);
+    processCameraProvider!
+        .bindToLifecycle(cameraSelector!, <UseCase>[videoCapture!]);
+    videoOutputPath =
+        await SystemServices.getTempFilePath(_videoPrefix, '.temp');
+    pendingRecording = await recorder!.prepareRecording(videoOutputPath!);
+    recording = await pendingRecording!.start();
+  }
+
+  /// Stops the video recording and returns the file where it was saved.
+  @override
+  Future<XFile> stopVideoRecording(int cameraId) async {
+    recording!.close();
```

Comment:
Hmm it looks like in the native side we return success but with null path, and then on the [dart side we throw an error](https://github.com/flutter/packages/blob/main/packages/camera/camera_android/lib/src/android_camera.dart#L280)? Will look into this because I'm not sure what the best behavior is given that the interface makes us return a non-null file.